app/src/main/java/com/spendster/presentation/addExpenses/AddExpensesActivity.java

```diff
@@ -130,5 +161,36 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             Log.d(BuildConfig.MY_LOGS, "Empty Data");
         }
     }
+
+    private void showCategoryInScreen(Intent data) {
```

Comment:
@SimonMaha maybe a better naming would be onCategorySelected. 
Also, this function is making two different things:
- parsing data from intent
- presenting data on UI

I would leave data parsing in onActivityResult and pass data to present here.